.github/workflows/cicd.yml

```diff
@@ -0,0 +1,185 @@
+# something more accurate once this is fixed.  See
+# <https://github.community/t/is-there-a-way-to-delete-or-hide-old-renamed-workflows/16281/15>
+name: "CI"
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ubuntu-18.04
+    strategy:
+      fail-fast: false
+      matrix:
+        emacs_version:
+          - 26.3
+          - snapshot
+    steps:
+      - uses: purcell/setup-emacs@master
+        with:
+          version: ${{ matrix.emacs_version }}
+
+      - uses: actions/checkout@v2
+
+      - name: Initialize sandbox
+        run: |
+          SANDBOX_DIR=$(mktemp -d) || exit 1
+          echo ::set-env name=SANDBOX_DIR::$SANDBOX_DIR
+          ./makem.sh -vv --sandbox=$SANDBOX_DIR --install-deps --install-linters
+
+      # The "all" rule is not used, because it treats compilation warnings
+      # as failures, so linting and testing are run as separate steps.
+
+      - name: Install JavaScript dependencies
+        run: npm install
+
+      - name: Install additional packages
+        run: sudo apt-get -y install aspell python3 python3-setuptools texinfo
+
+      - name: Install pandoc 2
+        run: |
+          curl -LO https://github.com/jgm/pandoc/releases/download/2.9.2.1/pandoc-2.9.2.1-1-amd64.deb
+          sudo dpkg -i pandoc-2.9.2.1-1-amd64.deb
+
+      - name: Install panflute
+        run: sudo pip3 install panflute
+
+      - name: Lint
+        run: ./makem.sh -vv --sandbox=$SANDBOX_DIR lint-checkdoc lint-compile lint-declare lint-package lint-regexps
+
+      - name: Set version suffix for PRs
+        if: github.event_name == 'pull_request'
+        run: |
+          echo ::set-env name=VERSION_SUFFIX::-4
+
+      - name: Build and Test
+        if: always() # Run test even if linting fails.
+        run: |
+          make && ./makem.sh -vv --sandbox=$SANDBOX_DIR test
+          tar cf prettier.tar -C dist .
+          ls -ld prettier.tar
+
+      - name: Upload build
+        if: matrix.emacs_version == '26.3'
+        uses: actions/upload-artifact@v2
+        with:
+          path: prettier.tar
+          name: prettier.tar
+
+  deploy:
+    needs: build
+    runs-on: ubuntu-18.04
+    if: github.event_name == 'push' && github.ref == 'refs/heads/master'
+    steps:
+      - name: Download build
+        uses: actions/download-artifact@v2
+        with:
+          name: prettier.tar
+
+      - name: Unpack tarball
+        run: |
+          tar xf prettier.tar --one-top-level=dist
+
+      - name: Get commit message
+        run: |
+          set -eou pipefail
+          echo ::set-env name=commit_message::$(
+            curl \
+              --silent \
+              --show-error \
+              --header "Accept: application/vnd.github.v3+json" \
+              --header "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+              "https://api.github.com/repos/${{ github.repository }}/commits/${{ github.sha }}" \
+            | jq --raw-output .commit.message
```

Comment:
Although I'm not familiar with Github Actions or don't run the deploy action, I think `sudo apt-get install jq` might be needed before using `jq`.

Reply:
Thanks for taking a close look. I've tested this in a private repository and it was fine, I assume the ubuntu base image already contains it. Going to merge this now, thanks again.